### PR TITLE
fix: handle nullish items

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/studio-internal/issues/1608

### CODE REVIEW

File: src/components/shared/Property.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface IProperty {
 export const Property: React.FunctionComponent<IProperty> = ({ node, onGoToRef }) => {
   const type = isRefNode(node) ? '$ref' : isCombinerNode(node) ? node.combiner : node.type;
   const subtype =
-    type === SchemaKind.Array && (node as IArrayNode).items !== undefined
+    type === SchemaKind.Array && _isObjectLike((node as IArrayNode).items)
```
Comment (marbemac) [inline src/components/shared/Property.tsx]: Should we start creating some type guards, like `isArrayNode`, and using those?

Any other spots where we might end up w unexpected null. `properties`, `anyOf`, `allOf`, `oneOf`, etc?

Comment (P0lip) [reply src/components/shared/Property.tsx]: > Should we start creating some type guards, like isArrayNode, and using those?

Yeah, sounds indeed a look a good idea. Will do.

> Any other spots where we might end up w unexpected null. properties, anyOf, allOf, oneOf, etc?

Yes there might be other spots, _but_ we take nulls into account in a few places, but just not all.

**Comment by P0lip** [comment]
Done
In general typings need some love in json-schema-viewer.
It became quite a bit of a hassle once we started addressing all the weirdness and quirkiness.
`chore/typings` branched my work off here, gonna polish it one day, mostly to utilize tagged union literals and being more careful about malformed data.
I believe we need to take a similar approach as in case of http-spec and assume invalid data is passed rather than the data meets our expectations.

**Comment by marbemac** [comment]
> I believe we need to take a similar approach as in case of http-spec and assume invalid data is passed rather than the data meets our expectations.

Or we add a transformer to `http-spec` package to "clean/normalize" json schema to make it valid, so that our entire ecosystem doesn't have to worry about these issues.

**Comment by P0lip** [comment]
> Or we add a transformer to http-spec package to "clean/normalize" json schema to make it valid, so that our entire ecosystem doesn't have to worry about these issues.

That would be some solution as well, but wondering how it'd impact the perf of large schemas processing. If the hit isn't significant, I'd be all in for bringing it to JSV, as right now we kind of need to do the same stuff twice.

**Pull request merged**